presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8186,6 +8186,22 @@ public void testLateralJoin()
                 "line .*: LATERAL on other than the right side of CROSS JOIN is not supported");
     }
 
+    @Test
+    public void testJoinsWithNulls()
+    {
+        assertQuery("SELECT * FROM (VALUES 2, 3, null) a(x) INNER JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x",
+                "SELECT * FROM VALUES (3, 3)");
+
+        assertQuery("SELECT * FROM (VALUES 2, 3, null) a(x) LEFT JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x",
+                "SELECT * FROM VALUES (3, 3), (2, NULL), (NULL, NULL)");
+
+        assertQuery("SELECT * FROM (VALUES 2, 3, null) a(x) RIGHT JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x",
+                "SELECT * FROM VALUES (3, 3), (NULL, 4), (NULL, NULL)");
+
+        assertQuery("SELECT * FROM (VALUES 2, 3, null) a(x) FULL OUTER JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x",
+                "SELECT * FROM VALUES (3, 3), (NULL, 4), (2, NULL), (NULL, NULL), (NULL, NULL)");
+    }
```

Comment:
Looks good. Just add one more test with some conditions like: a.x is null etc. so we can ensure that we did not push that down.

Reply:
Added now. It's interesting to see how NULL filtering works in FULL OUTER JOIN.

Reply:
I actually meant WHERE clauses not AND in ON clauses because we should not be pushing those down.

Reply:
@kaikalur Sure. My code does not touch Predicate pushdown. So this should not change. Updating test just for completeness.